Incompatible with the "node_modules" folder structure in `npm@3`

I ran into this issue today when I used `npm@3` to install  `grunt-not-constantinople`, which &mdash; as you know &mdash; has [`peerDependencies` on `grunt-contrib-clean`, `grunt-istanbul`, and `grunt-istanbul-coverage`](https://github.com/traviswimer/grunt-not-constantinople/blob/d236cff13b321b54bf1f0ddd3fb9bae9079cab04/tasks/not_constantinople.js#L44-L47):

```
Running "not_constantinople:coverage" (not_constantinople) task
Verifying property not_constantinople.coverage exists in config...OK
File: [no files]
Warning: Cannot find module 'grunt-contrib-clean' Use --force to continue.

Aborted due to warnings.
```

Realistically, the problem is in how `grunt.loadNpmTasks`/`grunt.task.loadNpmTasks` works... it is very specifically written to work in the `npm@<3` world as it relies on a `node_modules` directory existing as a child of the main directory rather than using a more robust mechanism like [`require.resolve`](https://nodejs.org/api/globals.html#globals_require_resolve) to traverse into module directories.

Ironically, with the current version of Grunt and `npm@3`, the [`grunt-loadNpmTasks`](https://www.npmjs.com/package/grunt-loadnpmtasks) Node module that I wrote a long time ago for a hierarchical codebase would now actually be applicable to a non-hierarchical codebase to solve this issue, too. :confused:

See https://github.com/gruntjs/grunt/issues/1355#issuecomment-175789630 for more background.

--- a/package.json
+++ b/package.json
@@ -15,6 +15,7 @@
   "bugs": {
     "url": "https://github.com/traviswimer/grunt-not-constantinople/issues"
   },
+  "license": "MIT",
   "licenses": [
     {
       "type": "MIT",
@@ -27,6 +28,15 @@
   "scripts": {
     "test": "grunt test"
   },
+  "dependencies": {
+    "grunt-loadnpmtasks": "^0.3.3"
+  },
+  "peerDependencies": {
+    "grunt": ">=0.4.0",
+    "grunt-istanbul": ">=0.2.0",
+    "grunt-istanbul-coverage": ">=0.0.2",
+    "grunt-contrib-clean": ">=0.4.0"
+  },
   "devDependencies": {
     "grunt-contrib-jshint": "~0.6.0",
     "grunt-contrib-nodeunit": "~0.2.0",
@@ -35,12 +45,6 @@
     "grunt-istanbul": "^0.2.4",
     "grunt-istanbul-coverage": "^0.0.2",
     "grunt-contrib-clean": "^0.4.0"
-  },
-  "peerDependencies": {
-    "grunt": ">=0.4.0",
-    "grunt-istanbul": ">=0.2.0",
-    "grunt-istanbul-coverage": ">=0.0.2",
-    "grunt-contrib-clean": ">=0.4.0"
   },
   "keywords": [
     "gruntplugin",

--- a/tasks/not_constantinople.js
+++ b/tasks/not_constantinople.js
@@ -10,13 +10,13 @@
 
 module.exports = function(grunt) {
 
+  var loader = require('grunt-loadnpmtasks')(grunt);
+
+
   // Please see the Grunt documentation for more information regarding task
   // creation: http://gruntjs.com/creating-tasks
 
   grunt.registerMultiTask('not_constantinople', 'Grunt task to easily add Istanbul code coverage using any unit-testing framework.', function() {
-
-
-    var cleaner = require('grunt-contrib-clean');
 
     // Merge task-specific and/or target-specific options with these defaults.
     var options = this.options({
@@ -42,9 +42,9 @@ module.exports = function(grunt) {
     var dirs = options.directories;
 
     // Load task dependencies
-    grunt.loadNpmTasks('grunt-contrib-clean');
-    grunt.loadNpmTasks('grunt-istanbul');
-    grunt.loadNpmTasks('grunt-istanbul-coverage');
+    loader.loadNpmTasks('grunt-contrib-clean');
+    loader.loadNpmTasks('grunt-istanbul');
+    loader.loadNpmTasks('grunt-istanbul-coverage');
 
     // Set clean config
     if( options.cleanup === true ){

